src/graph/graph.h

```diff
@@ -65,8 +65,8 @@ struct Graph {
 	GrB_Matrix *relations;              // Relation matrices.
 	GrB_Matrix *_relations_map;         // Maps from (relation, row, col) to edge id.
 	GrB_Matrix _zero_matrix;            // Zero matrix.
+	rax *_matrices_mutex;               // Matrices mutex map.
```

Comment:
Although convenient to have all locks under a single `rax` object, I'll prefer introducing a struct `Matrix` containing both `GrB_Matrix` and `pthread_mutex_t`
```
Matrix *relations;
Matrix *_relations_map;
Matrix _zero_matrix;
...
```

We might want to name that struct differently but you get my point.